rust/stub/src/main.rs

```diff
@@ -236,6 +243,30 @@ fn main(handle: Handle, mut system_table: SystemTable<Boot>) -> Status {
         warn!("Hash mismatch for initrd!");
     }
 
+    if tpm_available(system_table.boot_services()) {
+        debug!("TPM available, will proceed to measurements.");
+    }
+
+    if let Ok(features) = get_loader_features(system_table.runtime_services()) {
+        if features.contains(SystemdLoaderFeatures::RandomSeed) {
+            // FIXME: process random seed then on the disk.
+            debug!("Random seed is available, but lanzaboote does not support it yet.");
+        }
+    }
+
+    unsafe {
+        // Iterate over unified sections and measure them
+        let _ = measure_image(&system_table, booted_image_file(
+            system_table.boot_services()
+        ).unwrap())
+        .expect("Failed to measure the image");
```

Comment:
I'm not so sure about hard failing here.

Reply:
When we fail to parse our own memory as a PE for example, I'd say

Reply:
We don't want people doing strange things to the system making us fragile to security, so I'd say hardfailing here is not too shocking but… difficult to say.

Reply:
We can "not hard fail" then re-evaluate a "extreme stub mode" that tries to thwart active attacks.